content-src/components/DiscoveryStreamComponents/DSContextFooter/DSContextFooter.jsx

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { cardContextTypes } from "../../Card/types.js";
+import { CSSTransition, TransitionGroup } from "react-transition-group";
+import React from "react";
+
+const StatusMessage = ({ icon, fluentID }) => (
+  <div className="status-message">
+    <span
+      aria-haspopup="true"
+      className={`story-badge-icon icon icon-${icon}`}
+    />
+    <div className="story-context-label" data-l10n-id={fluentID} />
+  </div>
+);
+
+export class DSContextFooter extends React.PureComponent {
+  render() {
+    const { context, context_type } = this.props;
+    const { icon, fluentID } = cardContextTypes[context_type] || {};
+
+    return (
+      <div className="story-footer">
+        {context && <p className="story-sponsored-label">{context}</p>}
+        <TransitionGroup component={null}>
```

Comment:
we can move TransitionGroup inside !context check as we don't need it for sponsored stories

Reply:
This is true but we're somehow losing the first badge animation by doing this. My hunch is that putting this in the !context check makes React render TransitionGroup and CSSTransition at the sametime, where instead we would need TransitionGroup to preexist to render incoming and outgoing CSS transitions.

This doesn't render an extra div though if we set component={null}

Reply:
Alternatively... We can move sponsored context out of here (bring it back to DSCard) since in the Hero component the Sponsored Text is not in the footer but in the header.

https://github.com/mozilla/activity-stream/blob/20eb9c15127a2185d75fdc4358acee80a372f54b/content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx#L104-L114

Later PR w/ Hero & List
https://github.com/mozilla/activity-stream/pull/5206/commits/f90ef43060abebd8709550c8e456921d6842d041 using DSFooter